File: frontend/packages/types/src/weather.ts
/** Weather types (WEB-SEARCH-02A). */

export type WeatherCondition = string;

/**
 * Provenance de la reponse meteo (C3-D1.2-R3A).
 *
 * `development_stub` designe la preview locale deterministe du backend quand
 * aucune cle provider n'est configuree. Ses valeurs sont FABRIQUEES : aucun
 * consommateur ne doit les afficher comme une meteo reelle.
 */
export type WeatherSource = "provider" | "development_stub";

export interface WeatherCurrent {
  temperature: number;
  feels_like: number;
  condition: WeatherCondition;
  icon: string | null;
  sunrise: string | null;
  sunset: string | null;
  is_day: boolean;
  city: string;
  country: string | null;
  source: WeatherSource;
  /** Fournis uniquement par le provider reel ; absents sinon. */
  temperature_min?: number | null;
  temperature_max?: number | null;
  wind_speed?: number | null;
}

File: backend/app/schemas/weather.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

WeatherSource = Literal["provider", "development_stub"]


class WeatherCurrentOut(BaseModel):
    """Current weather for UI (WEB-SEARCH-02A).

    `source` (C3-D1.2-R3A) rend la provenance EXPLICITE : sans lui, le stub de
    developpement etait structurellement indiscernable d'une reponse
    OpenWeatherMap, et un consommateur pouvait afficher des degres fabriques
    comme s'ils etaient reels. Extension additive : les champs existants sont
    inchanges.
    """

    temperature: float
    feels_like: float
    condition: str
    icon: str | None
    sunrise: datetime | None
    sunset: datetime | None
    is_day: bool
    city: str
    country: str | None
    source: WeatherSource
    # Fournis uniquement par le provider reel ; jamais calcules ni devines.
    temperature_min: float | None = None
    temperature_max: float | None = None
    wind_speed: float | None = None
